### ISSUE TITLE
`pattern_matching_keywords` should not trigger in some cases

### ISSUE THREAD

**Issue by blindmonkey** [realm/SwiftLint#3852]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
`pattern_matching_keywords` opt-in rule will warn this type of code.
```swift
let z = 3
switch foo {
case (let x, let y, z): break
}
```

When matching a pattern it may be desirable to bind some elements of the tuple while comparing others to an existing variable. Currently, `pattern_matching_keywords` simply counts the number of `let`s and `var`s, causing it to miss cases like this.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? Verified via unit tests on master
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Built from source
* Paste your configuration file:

```yml
opt_in_rules:
  - pattern_matching_keywords
```

* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 13.2.1, Build version 13C100
